tests/certs/scripts/genserv.sh

```diff
@@ -59,6 +59,13 @@ if [ ".$SERIAL" = . ] ; then
 	SERIAL=`/usr/bin/env perl -e "$GETSERIAL"`
 fi
 
+# A hack for the expired certificate test case. We could do better, but it would
+# require using "openssl ca" instead of "openssl x509" for creating the server
+# certificates, as "openssl ca" lets you precisely specify start and end dates.
+if grep -qe "exp$" <<< "$PREFIX"; then
```

Comment:
I think `<<<` is a Bash-ism. Maybe better use `case`, which is available in all POSIX-compliant shells:

```sh
case "$PREFIX" in
*exp)
    DURATION=-$DURATION
    ;;
esac
```